### PR TITLE
Ability to run orawls as non-root user

### PR DESCRIPTION
**Pull request by yadavnikhil**
I'm contributing this changes on behalf of Amdocs Development Ltd.

Add functionality to run orawls with a non-root user.
This will work only with Puppet v3.7.0 or higher so as to use the patch for "exec" resource type which does not switch user if running user & target user are same & does not fail for switch user test. 
Following was done so as to have it running with non-root user:
1. Override default values for wls_domains.yaml & wls_setting.yaml from default /etc to custom location.
This is done by setting facter for overriding these defaults to custom location where the files will persist till the installation lifecycle:
export FACTER_override_wls_domains_file=<custom location>/wls_domains.yaml
export FACTER_override_wls_setting_file=<custom location>/wls_setting.yaml

If no facter is specified, it will be created at default location /etc

2. Issue: Only root can execute commands as other users
For user parameter in "exec" resource types, puppet has provided a patch from v3.7 onwards to validate & check user is root or not & does not fail if user to run with is same as running user which invoked puppet. 
So to minimize code changes for all exec resources, for now, orawls can be run by non-root user only with puppet v3.7+.

3. For bypassing easy_type running command with (su -), the code was modified to check if running user is root, else no user is passed to easy_type daemon.rb.

### CODE REVIEW

**Comment by biemond** [comment]
Very nice, I will check it out

👍

**Comment by biemond** [comment]
I had to change the domain.pp to make it work

**Comment by yadavnikhil** [comment]
Great, thanks a lot.

**Comment by ripun** [comment]
very Nice @yadavnikhil .. Thanks a lot Great work

**Comment by yapale** [comment]
This will be indeed very usefull

File: lib/utils/wls_access.rb [hunk 1]
```diff
@@ -324,16 +324,25 @@ def execute_wlst(script, tmpFile, parameters, domain, domainValues, options = {}
...
       debugmode = Puppet::Util::Log.level
+          
+      if Puppet.features.root?
+        eval_operatingSystemUser = operatingSystemUser
+      end
+      
       if debugmode.to_s == 'debug'
-        puts 'Prepare to run: ' + tmpFile.path + ',' +  operatingSystemUser + ',' +  domain + ',' +  weblogicHomeDir + ',' +  weblogicUser + ',' +  weblogicPassword + ',' +  weblogicConnectUrl
+        if eval_operatingSystemUser
+          puts 'Prepare to run: ' + tmpFile.path + ',' +  eval_operatingSystemUser + ',' +  domain + ',' +  weblogicHomeDir + ',' +  weblogicUser + ',' +  weblogicPassword + ',' +  weblogicConnectUrL
```
Comment (alexjfisher) [inline lib/utils/wls_access.rb]: `weblogicConnectUrL` typo?

Comment (alexjfisher) [inline lib/utils/wls_access.rb]: https://github.com/biemond/biemond-orawls/pull/353

**Pull request merged**